src/Shared/TypeExtensions.cs

```diff
@@ -105,4 +103,62 @@ private static bool IsGenericDefinition(this Type type, Type definition)
                    type.GetGenericTypeDefinition() == definition;
         }
     }
+
+    internal static class TypeHelper
+    {
+        public static Type GetUnderlyingTypeOrSelf(Type type)
+        {
+            return Nullable.GetUnderlyingType(type) ?? type;
```

Comment:
Great! I forgot to take nullable temporal types into account when comparing type in `ChangeSetPreparer` and the converters. I will change to use this helper class for other type comparisons later.